// .proto def for the querying service.

syntax = "proto3";
package ir.search;
//import "google/protobuf/struct.proto";

service IR {
  rpc QueryES(QueryRequest) returns (stream ResultEntry);
  rpc SignupUser(UserData) returns (UserId);
  rpc UserInfo(UserId) returns (UserData);
}

message QueryRequest{
  int64 id = 1;
  string query = 2;

}

message ResultEntry {
  string id = 1;
  double score = 2;
  map<string, string> data = 3;
}

message UserData {
  int64 UserId = 1;
  string name = 2;
  map<string, string> languages = 3;
  map<string, string> topics = 4;
  map<string, string> authors = 5;
  int64 nationality = 6;
  string sex = 7;
  int64 age = 8;
}

message UserId {
  int64 id = 1;
}